SRS Document.md

```diff
@@ -0,0 +1,112 @@
+|CI tools| software applications that assist the user in efficient integration and code development |
+
+
+# 2. Overall Description
+
+## 2.1 Product Perspective
+* **JavaScript**: With the support of the WebdriverIO library, the user builds the script.
+* **NodeJS**: NodeJS is an open-source project that makes the JavaScript runtime environment easier to use.
+* **WebdriverIO**: WebdriverIO is a NodeJS-based application that interacts with NodeJS.
+
+![](https://github.com/TemirlanAidarov/Images/blob/main/Diagram.drawio.svg)
+
+
+WebdriverIO is based on NodeJS, which is considered to be a JSON Wire Protocol implementation. It performs automated testing using a RESTful architecture.
+
+With the support of the WebdriverIO library, the user builds the script in JavaScript. In that case, the Service request is transmitted through NodeJS as an HTTP instruction. It appllies the JSON Wire protocol and the service module redirects the request to the webbrowser.
+
+Then the browser executes the user actions after getting the instruction, ensuring that the application functionalities are legitimate.
+
+## 2.2 Product Functions
+Using this WebdriverIO, a user can automate modern web written in  React, Angular, Polymeror Vue.js as well as native mobile applications for Android and iOS.
+
+* WebDriver Protocol (for cross browser testing)
+* Chrome Dev Tools Protocol (for chromium based automation)
+* Appium (for mobile automation)
+
+## 2.3 User Classes and Characteristics
+In general, WebdriverIO is characterized as a simple and accessible tool for creating web and mobile app tests. It is accessible to anyone who wants to learn and write frontend tests in JavaScript (TypeScript) and needs some assistance.
+
+WebdriverIO is easy to start and get used to for beginners. The installation process is straightforward and quick. Users do not need to install any browser drivers and configure themselves. After installation, the user only needs to write a config command that will perform all the necessary configuration. Further details are described in the [documentation page](https://webdriver.io/docs/gettingstarted) 
+
+## 2.4 Operating Environment
+A user has to install Node.js. It is suggested to choose the *LTS* version, so it is suggested to choose 12.16.1 or above. 
+There are no other strict rules regarding system requirements. Nevertheless, in order to work with little problem 
+as possible with Node.js and WebdriverIO, it is advised for users to have:
+* Ubuntu version 16.04 or higher for Linux 
+* or Windows 7 or newer versions with at least 2GB RAM (4GB preferable). 
+
+## 2.5 Assumptions and Dependencies
+The WebdriverIO community has to build a full test framework with various additional features (e.g. making the CLI tool more powerful, Jest Framework Support). Still, the WebdriverIO community is active and working on continuous improvements. 
```

Comment:
We don't support the Jest framework (note that you can run WebdriverIO in all frameworks but the testrunner only supports Mocha, Jasmine and Cucumber as test frameworks). We allow however to write tests in the same way as Jest tests are written.